compiler/compile/OMRCompilation.hpp

```diff
@@ -676,6 +676,36 @@ class OMR_EXTENSIBLE Compilation
 
    void setIlVerifier(TR::IlVerifier *ilVerifier) { _ilVerifier = ilVerifier; }
 
+   typedef std::pair<void *, TR::DebugCounterBase *> DebugCounterEntry;
+   typedef TR::typed_allocator<DebugCounterEntry, TR::Allocator> DebugCounterMapAllocator;
+   typedef std::map<void *, TR::DebugCounterBase *, std::less<void *>, DebugCounterMapAllocator> DebugCounterMap;
+
+   /**
+    * @brief mapStaticAddressToCounter
+    * @param staticAddress the static address of the memory used to store the count value of the debug counter
+    * @param counter pointer to the debug counter
+    *
+    * Maps the staticAddress to the debug counter. The map has to be between the static address and the
+    * debug counter, instead of the symref of the static address and the debug counter, because sometimes,
+    * transformations can create a new symref that will not exist in this map; however because the static
+    * address is constant, it is guaranteed to exist in the map.
+    */
+   void mapStaticAddressToCounter(void *staticAddress, TR::DebugCounterBase *counter);
```

Comment:
Do you ever need this two functions in a place where you do not have the symbol reference?

If not, then I suggest changing the API to take a symbol reference, but assert if it isn't a static symbol reference and dig out the static address here as the key for the map.

Reply:
> Do you ever need this two functions in a place where you do not have the symbol reference?

No, I believe in every place that this function is called, I get the static address from the symref. I'll update the code as per your suggestion.